### PR TITLE
nomad-exec: pick allocation by supplied tag

### PR DESCRIPTION
**Pull request by nrdxp**
If no tag matches, the script will still fall back to the picker.

When combined with `-j` and `-t` flags, this should allow the script to now function non-interactively.

Only open question, should we match exactly (`==`) or with regex (`~=`)? This currently implements the former.

### CODE REVIEW

**Comment by blaggacao** [comment]
> Only open question, should we match exactly (==) or with regex (~=)? This currently implements the former.

Definitely exact matches. Especially since this may be used in scripting.

File: nomad-exec.sh [hunk 1]
```diff
@@ -31,6 +32,7 @@ while getopts 'c:u:n:j:t:a:e:psvh' c; do
     echo '  -e  CMD for exec. Default: "/bin/debug"'
     echo "  -p  prompt for some defaults if not declared as options (NOMAD_NAMESPACE, CMD)"
     echo "  -s  show consul service allocation associated addresses and tags (requires services to be tagged with the NOMAD_ALLOC_ID)"
+    echo "  -S  Choose the first allocation matching the given consul tag (implies '-s')"
```
Comment (johnalotoski) [inline nomad-exec.sh]: Suggest same format as above here:
```
-S  CONSUL_TAG to choose the first allocation matching the given consul tag (implies '-s')"
```

Comment (nrdxp) [reply nomad-exec.sh]: done

**Review by johnalotoski — APPROVED** [review]

**Comment by blaggacao** [comment]
Thanks @nrdxp for the implementation!
Thanks @johnalotoski for the review!

**Pull request merged**